android/java/org/chromium/chrome/browser/vpn/utils/InAppPurchaseWrapper.java

```diff
@@ -178,8 +179,10 @@ private PurchasesUpdatedListener getPurchasesUpdatedListener(Context context) {
                              Toast.LENGTH_SHORT)
                         .show();
             } else if (billingResult.getResponseCode()
-                    == BillingClient.BillingResponseCode.SERVICE_DISCONNECTED) {
+                            == BillingClient.BillingResponseCode.SERVICE_DISCONNECTED
+                    && retryCount < 5) {
                 connectToBillingService();
+                retryCount++;
```

Comment:
do we need to set it to 0 after success or that class gets re-created every time?